pkg/kubectl/sorting_printer.go

```diff
@@ -153,11 +154,56 @@ func isLess(i, j reflect.Value) (bool, error) {
 		return i.String() < j.String(), nil
 	case reflect.Ptr:
 		return isLess(i.Elem(), j.Elem())
+	case reflect.Struct:
+		// special case handling, if the struct has methods: Before, etc
+		// TODO: add more?
+		supported, less := lessFunc(i, j, "Before")
+		if supported {
+			return less, nil
+		}
+		// fallback for the fields comparision
+		for idx := 0; idx < i.NumField(); idx++ {
+			less, err := isLess(i.Field(idx), j.Field(idx))
+			if err != nil || !less {
+				return less, err
+			}
+		}
+		return true, nil
+	case reflect.Array, reflect.Slice:
+		// note: the length of i and j may be different
+		for idx := 0; idx < integer.IntMin(i.Len(), j.Len()); idx++ {
+			less, err := isLess(i.Index(idx), j.Index(idx))
+			if err != nil || !less {
+				return less, err
+			}
+		}
+		return true, nil
 	default:
 		return false, fmt.Errorf("unsortable type: %v", i.Kind())
 	}
 }
 
+// this is a helper func to compare two structs when the given methods exists
+func lessFunc(i, j reflect.Value, methods ...string) (supported, less bool) {
+	for _, name := range methods {
+		f := i.MethodByName(name)
+		fType := f.Type()
+		// it must be a func and take a single argument which has the same type as j
+		// also it must returns a bool value
```

Comment:
It is specific to timestamps for now given the `Before` method name.

I am hoping the method signature would be something like `func (a *A) name(b *A) bool`, the name could be `Before`, or anything else(I added a TODO comment above to add more such names when needed), it does not look good though, but as far as I can tell, it may be fine for now or do you have any suggestions on the method signature which could be used here?